dataprocessors/json/json.go

```diff
@@ -0,0 +1,137 @@
+package json
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/spiceai/data-components-contrib/dataprocessors/json/observation"
+	"github.com/spiceai/spiceai/pkg/observations"
+	"github.com/spiceai/spiceai/pkg/state"
+	"github.com/spiceai/spiceai/pkg/util"
+	"go.skia.org/infra/go/jsonschema"
+)
+
+const (
+	JsonProcessorName string = "json"
+)
+
+type JsonProcessor struct {
+	data      []byte
+	dataMutex sync.RWMutex
+	dataHash  []byte
+	format    JsonFormat
+}
+
+type JsonFormat interface {
+	GetSchema() *[]byte
+	GetObservations(data *[]byte) ([]observations.Observation, error)
+	GetState(data *[]byte, validFields *[]string) ([]*state.State, error)
```

Comment:
Don't need to pass maps/arrays by pointers unless you want the func to be able to replace it.